lib/workup/application.rb

```diff
@@ -47,53 +46,11 @@ def initialize(*args)
       super(*args)
     end
 
-    no_commands do
-      def chef_lib(description)
-        log.info description
-        return_code = Workup::Helpers.silence { yield }
-        if return_code.zero?
-          log.debug "OK\n"
-        else
-          log.error "Failure\n"
-          exit return_code
-        end
-      end
-    end
-
-    desc 'default', 'Default task'
-    def default
-      Workup::Helpers.prompt_for_password if options[:password]
-      chef_zero
-      workup
-    end
-
-    desc 'chef_zero', 'Create the chef-zero directory'
-    def chef_zero
-      Workup::Helpers.initialize_files(options[:workup_dir])
-
-      policyfile = options[:policyfile]
-      policyfile = File.join(Dir.pwd, policyfile) if Pathname.new(policyfile).relative?
-
-      chefzero_path = File.join(options[:workup_dir], 'chef-zero')
-
-      chef_lib('Updating lock file... ') do
-        ChefDK::Command::Update.new.run([policyfile])
-      end
-
-      chef_lib('Creating chef-zero directory... ') do
-        ChefDK::Command::Export.new.run(['--force', policyfile, chefzero_path])
-      end
-    end
-
-    desc 'workup', 'Run workup'
+    desc 'workup', 'Run Workup'
     def workup
-      log.info "Starting workup\n"
-      Workup::Helpers.initialize_files(options[:workup_dir])
-      Workup::Helpers.prompt_for_password if options[:password]
-      Workup::Helpers.chef_client(File.join(options[:workup_dir], 'client.rb'),
-                                  options[:dry_run])
+      Workup::Runner.new(log, options).run
```

Comment:
Moved everything that has to do with running workup into this script (probably should make similar objects for the workup workspace / move the helpers into those appropriate classes (for example chefdk functions that run chef should move from helpers into the runner. Or whatnot